README.rst

```diff
@@ -0,0 +1,94 @@
+Timeout decorator
+=================
+
+|Build Status| |Pypi Status| |Coveralls Status|
+
+Installation
+------------
+
+From source code:
+
+::
+
+    python setup.py install
+
+From pypi:
+
+::
+
+    pip install timeout-decorator
+
+Usage
+-----
+
+::
+
+    import time
+    import timeout_decorator
+
+    @timeout_decorator.timeout(5)
+    def mytest():
+        print "Start"
+        for i in range(1,10):
+            time.sleep(1)
+            print "%d seconds have passed" % i
+
+    if __name__ == '__main__':
+        mytest()
+
+Multithreading
+--------------
+
+By default, timeout-decorator uses signals to limit the execution time
+of the given function. This appoach does not work if your function is
+executed not in a main thread (for example if it's a worker thread of
+the web application). There is alternative timeout strategy for this
+case - by using multiprocessing. To use it, just pass
+``use_signals=False`` to the timeout decorator function:
+
+::
+
+    import time
+    import timeout_decorator
+
+    @timeout_decorator.timeout(5, use_signals=False)
+    def mytest():
+        print "Start"
+        for i in range(1,10):
+            time.sleep(1)
+            print "%d seconds have passed" % i
+
+    if __name__ == '__main__':
+        mytest()
+
+.. warning::
+    Make sure that in case of multiprocessing strategy for timeout, your function does not return objects which cannot
+    be pickled, otherwise it will fail at marshalling it between master and child processes.
+
+
+Acknowledgement
+---------------
+
+Derived from
+http://www.saltycrane.com/blog/2010/04/using-python-timeout-decorator-uploading-s3/
+and https://code.google.com/p/verse-quiz/source/browse/trunk/timeout.py
```

Comment:
Hm it's something on the dying Google code. I'll check.
On Mar 16, 2015 7:15 PM, "Patrick" <notifications@github.com> wrote:

> In README.rst
> <https://github.com/pnpnpn/timeout-decorator/pull/16#discussion_r26512185>
> :
>
> > +            print "%d seconds have passed" % i
> > +
> > +    if __name__ == '__main__':
> > +        mytest()
> > +
> > +.. warning::
> > +    Make sure that in case of multiprocessing strategy for timeout, your function does not return objects which cannot
> > +    be pickled, otherwise it will fail at marshalling it between master and child processes.
> > +
> > +
> > +Acknowledgement
> > +---------------
> > +
> > +Derived from
> > +http://www.saltycrane.com/blog/2010/04/using-python-timeout-decorator-uploading-s3/
> > +and https://code.google.com/p/verse-quiz/source/browse/trunk/timeout.py
>
> What is this? I am getting a 403 forbidden for this url.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pnpnpn/timeout-decorator/pull/16/files#r26512185>.
>